### ISSUE TITLE
Using without Django

### ISSUE THREAD

**Issue by robvdl** [miki725/django-url-filter#63]
I have plenty of SQLAlchemy code bases that are not based on Django where this library could be useful.

It looks like the sqlalchemy backend itself can be used without Django as I don't see any references to Django imports:

http://django-url-filter.readthedocs.io/en/latest/_modules/url_filter/backends/sqlalchemy.html

Have you considered splitting this part out to a separate library? For now I would have to copy this file in my codebase (keeping the comments + copyright reference off course), but it would be nice not to have to do this in the future.

The requirements.txt pulls in Django as a dependency that is why I can't exactly use it, it would basically trigger it installing Django in a flask project, or bottle project etc.

**Comment by miki725** [comment]
as a side note, dont believe importing forms require Django settings to initialize anywhere (or if it does will require very minimal Django settings module) so actually you could end up using it in flask project provided you are ok pulling Django as a dep even though 99% of it will never get utilized...

**Comment by robvdl** [comment]
Thank for the explanation, I might look at that or I might also look how much you are using Django forms for validation, if it is easy enough to replace that with my current validation library for example, marshmallow I might look into this at some point.

Mostly I am interested in the code that builds the double underscore syntax to the equivalent SQLAlchemy query.

I have already got my own basic code already that handles __gte, __lte, __gt, and __lt but I would like to do more like joins as well in the future.

I have stumbled upon such code before (can't remember the source now) but it was a bit invasive in that it liked to completely replace SQLAlchemy's query object and filter method. I like the idea of having that separate instead.
